examples/broadcasting_your_own_methods.py

```diff
@@ -0,0 +1,472 @@
+# ``DataArray`` that comes in will have the same structure.
+
+# %%
+# Making our Function Broadcastable
+# ---------------------------------
+# To combat this problem, we can make the observation that given any
+# ``DataArray``, we always want to broadcast our ``in_slippery_region``
+# function
+# along the ``"space"`` dimension. By "broadcast", we mean that we always want
+# to run our function for each 1D-slice in the ``"space"`` dimension, since
+# these are the (x, y) coordinates. As such, we can decorate our function with
+# the ``make_broadcastable`` decorator:
+
+
+@make_broadcastable()
+def in_slippery_region_broadcastable(xy_position) -> float:
+    return in_slippery_region(xy_position=xy_position)
+
+
+# %%
+# Note that when writing your own methods, there is no need to have both
+# ``in_slippery_region`` and ``in_slippery_region_broadcastable``, simply apply
+# the ``make_broadcastable`` decorator to ``in_slippery_region`` directly.
+# We've made two separate functions here to illustrate what's going on.
+
+# %%
+# ``in_slippery_region_broadcastable`` is usable in exactly the same ways as
+# ``in_slippery_region`` was:
+
+for point in [(0, 100), (450, 700), (550, 1500), (601, 500)]:
+    print(
+        f"{point} is in slippery region: "
+        f"{in_slippery_region_broadcastable(point)}"
+    )
+
+
+# %%
+# However, ``in_slippery_region_broadcastable`` also takes a ``DataArray`` as
+# the first (``xy_position``) argument, and an extra keyword argument
+# "``broadcast_dimension``". These arguments let us broadcast across the given
+# dimension of the input ``DataArray``, treating each 1D-slice as a separate
+# input to ``in_slippery_region``.
+
+in_slippery_region_broadcasting = in_slippery_region_broadcastable(
+    positions,  # Now a DataArray input
+    broadcast_dimension="space",
+)
+
+print("DataArray output using broadcasting: ")
+print(in_slippery_region_broadcasting)
+
+# %%
+# Calling ``in_slippery_region_broadcastable`` in this way gives us a
+# ``DataArray`` output - and one that retains any information that was in our
+# original ``DataArray`` to boot! The result is exactly the same as what we got
+# from using our ``for`` loop, and then adding the extra information to the
+# result.
+
+# Throws an AssertionError if the two inputs are not the same
+xr.testing.assert_equal(
+    was_in_slippery_region, in_slippery_region_broadcasting
+)
+
+# %%
+# But importantly, ``in_slippery_region_broadcastable`` also works on
+# ``DataArrays`` with different dimensions.
+# For example, we could have pre-selected one of our individuals beforehand.
+i_id = "AEON3B_NTP"
+individual_0 = positions.sel(individuals=i_id)
+
+individual_0_in_slippery_region = in_slippery_region_broadcastable(
+    individual_0,
+    broadcast_dimension="space",
+)
+
+print(
+    "We get a 3D DataArray output from our 4D input, "
+    "again with the 'space' dimension that we broadcast along collapsed:"
+)
+print(individual_0_in_slippery_region)
```

Comment:
To get an interactive html widget
```suggestion
individual_0_in_slippery_region
```